apps/ui/src/components/Header.tsx

```diff
@@ -17,49 +17,184 @@
  * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
-import { Button, ConfigProvider, Flex, Image, Layout, Typography } from 'antd';
+import { Button, ButtonProps, ConfigProvider, Drawer, Flex, Image, Layout, Typography, theme } from 'antd';
 
 import PCGL from '@/assets/pcgl-logo-full.png';
 import { pcglHeaderTheme } from '@/components/providers/ThemeProvider';
+import { Breakpoints, useMinWidth } from '@/global/hooks/useMinWidth';
+import { CloseOutlined, MenuOutlined } from '@ant-design/icons';
+import React, { useState } from 'react';
 
 const { Link } = Typography;
 const { Header } = Layout;
+const { useToken } = theme;
+
+interface MenuLinkType {
+	name: string;
+	isButton?: boolean;
+	buttonProps?: ButtonProps;
+	onClickAction?: VoidFunction;
+	href?: string;
+	position: 'left' | 'right';
+}
 
 const linkStyle: React.CSSProperties = {
 	minWidth: 100,
 	textAlign: 'center',
 };
 
 const HeaderComponent = () => {
+	const minWidth = useMinWidth();
+	const isResponsiveMode = minWidth <= Breakpoints.XL;
+
+	const { token } = useToken();
+
+	const [responsiveMenuOpen, setResponsiveMenuOpen] = useState(false);
+
+	const menuButtonStyle: React.CSSProperties = {
+		width: isResponsiveMode ? '100%' : 'auto',
+	};
+
+	/**
+	 * Default action when a button in the menu is clicked, used particularly for the mobile menu which should close after click.
+	 * @param buttonAction The function for the action needed to be preformed.
+	 */
+	const onMenuButtonClick = (buttonAction?: VoidFunction) => {
+		if (!buttonAction) {
+			return;
```

Comment:
My first reaction is that if `onMenuButtonClick` instantly returns when `buttonAction` is `undefined` then this argument shouldn't be optional and I would revise the implementation of onMenuButtonClick
Just reviewing what that might look like

Reply:
Ah, yeah theres a bit of weirdness there that I don't like either. The main reason it's like this is because `MenuLinkType` doesn't require you to give an `onClickAction` since usually you're just making links. If there's a better way to handle this let me know! 

```TS
interface MenuLinkType {
	name: string;
	isButton?: boolean;
	buttonProps?: ButtonProps;
	onClickAction?: VoidFunction;
	href?: string;
	position: 'left' | 'right';
}
```

Reply:
I would separate MenuLink into multiple types; a base MenuItem type with name and position, then a MenuLink and MenuButton type which extend MenuItem.

At first I was thinking that's overkill for one component, but come to think of it we will probably have more menus and can possibly reuse these definitions in the future.

Reply:
Similar thing with `href` -- every Link should have an href, but some items are buttons and some are links, so it's optional.

Reply:
This makes a lot of sense! I've gone ahead and refactored the code in commit 4586b8d as well as added an additional prop for anchor tag targets into `MenuLink` for more future extensibility.